### DISCUSSION TITLE
`wire:model` not work with alpinejs `x-model`

### DISCUSSION THREAD

**Discussion by mr-chetan** [livewire/livewire#9114] [7. Bugs (archived)] [answered]
### Livewire version

3.5.18

### Laravel version

11.36.1

### Which PHP version are you using?

PHP 8.4

### What is the problem?

Live Model does not Update if I enter text on the second input but if I enter text on the first input it will update the Livewire model

### Code snippets

```blade
    <div class="" x-data="{
        text: '',
    }">
        <p>{{ $text }}</p>
        <br>
        <input type="text" x-model="text" wire:model.live="text" />
        <input type="text" x-model="text"/>
    </div>
```

### Wirebox URL

https://wirebox.app/b/4z6rg

### How do you expect it to work?

I need to be update the livewire text

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
it's not how `wire:model` works under the hood (and it's not bug, see https://github.com/livewire/livewire/blob/62f0fa6b340a467c25baa590a567d9a134b357da/js/directives/wire-model.js). if you need to bind the model from alpine `x-data` and `wire:model`, you need to use `$wire.$entangle` and make it live update to re-render the view.

```blade
<div class="" x-data="{
    text: $wire.$entangle('text', live = true),
}">
    <p>{{ $text }}</p>
    <br>
    <input type="text" x-model="text" wire:model.live="text" />
    <input type="text" x-model="text" />
</div>
```

**Reply by mr-chetan** [reply]
can you give example with `$wire.$entangle`

**Reply by danie-ramdhani** [reply, marked as answer]
```blade
<div class="" x-data="{
    text: $wire.$entangle('text', live = true),
}">
    <p>{{ $text }}</p>
    <br>
    <input type="text" x-model="text" wire:model.live="text" />
    <input type="text" x-model="text" />
</div>
```

is the example above is not enough??
docs: https://livewire.laravel.com/docs/alpine#sharing-state-using-wireentangle
